src/components/primitives/Radio/Radio.tsx

```diff
@@ -23,85 +27,104 @@ const Radio = ({ icon, children, ...props }: IRadioProps, ref: any) => {
     ...props,
   });
 
+  const inputRef = React.useRef(null);
+  let { inputProps } = useRadio(props, contextState.state, inputRef);
+
   // only calling below function when icon exist.
   const sizedIcon = () =>
     //@ts-ignore
     React.cloneElement(icon, {
       size,
       color:
         //@ts-ignore
-        icon.props.color ?? checked
-          ? isDisabled
```

Comment:
I am not sure if we are handling isDisabled inside useRadio? If not then we will have to manually convert it from isDisabled to disabled.
Because isDisabled is a prop, while disabled is not.

Reply:
Yes, isDisabled is handled from useRadio hook. It sends disabled inside inputProps